modules/rostests/apitests/user32/keybd_event.c

```diff
@@ -0,0 +1,35 @@
+#include "precomp.h"
+
+void testScancodeExtendedKey(BYTE wVk, BYTE scanCode)
+{
+	trace("wVK: %x\tScancode: %x\n", wVk, scanCode);
+
+	keybd_event(0, scanCode, KEYEVENTF_SCANCODE | KEYEVENTF_EXTENDEDKEY, 0);
+    SHORT winKeyState = GetAsyncKeyState(wVk);
+    ok(winKeyState & 0x8000, "VK=%x should be detected as key down.\n", wVk);
+
+    keybd_event(0, scanCode, KEYEVENTF_SCANCODE | KEYEVENTF_EXTENDEDKEY | KEYEVENTF_KEYUP, 0);
+    winKeyState = GetAsyncKeyState(wVk);
```

Comment:
So, looking at `ProcessKeyEvent`, the async key state does seem to be a good way to tell what's happening in `UserSendKeyboardInput`. A low-level keyboard hook would be a slightly better test, because it would show a 1:1 reflection of what's happening in input processing, and the keycode passed to it doesn't go through `IntSimplifyVk`.
Similarly, `SendInput` would make a slightly better test for this than `keybd_event`, because it goes through less abstraction. With the test as is, I could ask "How do you know that it's not some magic inside `keybd_event` and/or `GetAsyncKeyState` that's making this test behave this way.
However, I think it's close enough to reasonable proof that your change is in the correct place -- now let's just make sure it doesn't cause any regressions.